spring-kafka/src/test/java/org/springframework/kafka/core/DefaultShareConsumerFactoryTests.java

```diff
@@ -197,4 +185,112 @@ void integrationTestDefaultShareConsumerFactory(EmbeddedKafkaBroker broker) thro
+	 * and returns a map of consumerId to the set of record values received by that consumer.
+	 */
+	private static Map<String, Set<String>> runSharedConsumerTest(String topic, String groupId,
+			List<String> consumerIds, int recordCount, EmbeddedKafkaBroker broker) throws Exception {
+		var bootstrapServers = broker.getBrokersAsString();
+
+		var producerProps = new java.util.Properties();
+		producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+		producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+		producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+		try (var producer = new KafkaProducer<String, String>(producerProps)) {
+			for (int i = 0; i < recordCount; i++) {
+				producer.send(new ProducerRecord<>(topic, "key" + i, topic + "-value-" + i)).get();
+			}
+			producer.flush();
+		}
+
+		setShareAutoOffsetResetEarliest(bootstrapServers, groupId);
+
+		Map<String, Set<String>> consumerRecords = new java.util.concurrent.ConcurrentHashMap<>();
+		consumerIds.forEach(id -> consumerRecords.put(id,
+				java.util.Collections.synchronizedSet(new java.util.HashSet<>())));
+		var latch = new java.util.concurrent.CountDownLatch(recordCount);
+		var running = new java.util.concurrent.atomic.AtomicBoolean(true);
+		ExecutorService executor = Executors.newCachedThreadPool();
+		List<Future<?>> futures = new java.util.ArrayList<>();
+
+		// Consumer task: poll, acknowledge, and count down latch for new records
+		for (int i = 0; i < consumerIds.size(); i++) {
+			final int idx = i;
+			Future<?> future = executor.submit(() -> {
+				DefaultShareConsumerFactory<String, String> shareConsumerFactory = new DefaultShareConsumerFactory<>(
+						Map.of(
+							"bootstrap.servers", bootstrapServers,
+							"key.deserializer", org.apache.kafka.common.serialization.StringDeserializer.class,
+							"value.deserializer", org.apache.kafka.common.serialization.StringDeserializer.class
+						));
+				try (var consumer = shareConsumerFactory
+						.createShareConsumer(groupId, consumerIds.get(idx))) {
+					consumer.subscribe(Collections.singletonList(topic));
+					while (running.get() && latch.getCount() > 0) {
+						var records = consumer.poll(Duration.ofMillis(200));
+						for (var r : records) {
+							if (consumerRecords.get(consumerIds.get(idx)).add(r.value())) {
+								consumer.acknowledge(r, AcknowledgeType.ACCEPT);
+								latch.countDown();
+							}
+						}
+					}
+				}
+			});
+			futures.add(future);
+		}
+
+		boolean completed = latch.await(10, java.util.concurrent.TimeUnit.SECONDS);
+		running.set(false);
+		for (Future<?> future : futures) {
+			try {
+				future.get();
+			}
+			catch (Exception e) {
+				throw new RuntimeException(e);
+			}
+		}
+		executor.shutdown();
+		assertThat(completed)
+			.as("All records should be received within timeout")
+			.isTrue();
```

Comment:
We don't need all of these extra variables and even don't need to deal with `Future`.
The `executor.shutdown();` and then assert on the `awaitTermination()` should do the trick.
See that `shutdown()` JavaDocs.